src/main/java/org/jpy/PyLib.java

```diff
@@ -391,11 +400,119 @@ private static void preloadPythonLib(String pythonLibPath) {
     private PyLib() {
     }
 
+    private static int randomInt = (int) (Math.random() * 1000000);
+
     static {
         if (DEBUG) System.out.println("org.jpy.PyLib: entered static initializer");
+        tryAutoConfiguration();
         loadLib();
         if (DEBUG) System.out.println("org.jpy.PyLib: exited static initializer");
     }
+
+    private static void tryAutoConfiguration()     {
+        if (DEBUG) System.out.println("org.jpy.PyLibConfig: trying to use C libraries from jar file.");
+
+        File jpySoFile = extractToTempFile(JPY_SO_FILE);
+        if (DEBUG) System.out.println("org.jpy.PyLibConfig: jpy.so file: " + jpySoFile.getAbsolutePath());
+        File jdlSoFile = extractToTempFile(JDL_SO_FILE);
+        if (DEBUG) System.out.println("org.jpy.PyLibConfig: jdl.so file: " + jdlSoFile.getAbsolutePath());
+        if(jpySoFile != null && jdlSoFile != null) {
+            File jpyConfigFile = createTempConfigFile(jpySoFile, jdlSoFile);
+            if (DEBUG) System.out.println("org.jpy.PyLibConfig: Setting property "
+                    + JPY_CONFIG_KEY + " to " + jpyConfigFile.getAbsolutePath() + ".");
+            System.setProperty(JPY_CONFIG_KEY, jpyConfigFile.getAbsolutePath());
+        }
+
+
+        if (DEBUG) System.out.println("org.jpy.PyLibConfig: End auto configuration.");
+
+    }
+
+    private static File extractToTempFile(String internalFilename) {
+        File filename = null;
+        InputStream fileStream = Thread.currentThread().getContextClassLoader().getResourceAsStream(internalFilename);
+        if (fileStream != null) {
+            OutputStream output = null;
+            byte[] buffer = new byte[8 * 1024];
+            try {
+                filename = new File(System.getProperty("java.io.tmpdir"), "jpy-" + randomInt);
```

Comment:
There is still a risk that this line overwrites an existing file. A safer alternative would be to delete `randomInt` and use `File.createTempFile("jpy-", ".so")` (or `."dll"`) instead.